pkg/api/metrics.go

```diff
@@ -40,6 +44,101 @@ func (hs *HTTPServer) QueryMetricsV2(c *models.ReqContext) response.Response {
 	return toJsonStreamingResponse(resp)
 }
 
+func checkDashboardAndPanel(ctx context.Context, dashboardId, panelId string) error {
+	if dashboardId == "" || panelId == "" {
+		return models.ErrDashboardOrPanelIdentifierNotSet
+	}
+
+	id, err := strconv.ParseInt(dashboardId, 10, 64)
+	if err != nil {
+		return models.ErrDashboardIdentifierInvalid
+	}
+
+	query := models.GetDashboardQuery{
+		Id: id,
+	}
+
+	if err := bus.Dispatch(ctx, &query); err != nil {
+		return err
+		//if errors.Is(err, models.ErrDataSourceNotFound) {
+		///return response.Error(404, "Data source not found", nil)
+		//}
+		//if errors.Is(err, models.ErrDataSourceIdentifierNotSet) {
+		//return response.Error(400, "Datasource id is missing", nil)
+		//}
+		//return response.Error(500, "Failed to query datasources", err)
+	}
+
+	// Dashboard has no properties. This would be weird, haven't written a test
+	// for this yet, but would most likely be a bug and return an error from
+	// bus.Dispatch
+	if query.Result == nil {
```

Comment:
Do we recover from panics in calling code here? Even if unlikely seems like the behavior of panicing the process would undesirable.

Reply:
We're not recovering from a panic here currently. Our thought process was that if this ever actually happens we have a serious problem and should shutdown. 

Happy to return an error and keep running if you think that makes more sense.

Reply:
I think better to change it given that Grafana is still largely a monolith so recycle time can be a bit. And if there was a dashboard making a bunch of validate queries might take out multiple instances depending on the lb setup.